pyccl/ccl_core.i

```diff
@@ -4,6 +4,18 @@
 /* put additional #include here */
 %}
 
+// SWIG black magic
```

Comment:
OK, I'm not familiar with this at all. Can you explain or point me to a reference?

Reply:
Well. There's http://www.swig.org/Doc3.0/SWIGDocumentation.html#Typemaps but that's not particularly useful. My very limited understanding is that typemaps hook into the C<-> python interface to deal with type conversions etc. The `memberin` typemap hooks into actions with structure members. In this case it thows an exception when there's and input.

I tried all afternoon to get things like %extend or %immutable to work, or overriding the setter functions but none of that seemed to do anything. I was sceptical about SWIG before but now I feel like converting everything to something else like pybind might not be the worst idea.